scrapper.py

```diff
@@ -25,39 +38,135 @@ class Crawler:
+        self.seed_urls = seed_urls
+        self.max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        content_bs = article_bs.find_all('div', class_="issueArticle flex")
+        for tag in content_bs:
+            if len(self.urls) >= self.max_articles:
+                break
+            link = tag.find('a')
+            main_link = link['href']
+            self.urls.append(main_link)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for seed_url in self.seed_urls:
+            time.sleep(1)
+            response = requests.get(seed_url, HEADERS)
+            article_bs = BeautifulSoup(response.text, 'html.parser')
+            self._extract_url(article_bs)
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    path = Path(base_path)
+    if path.is_dir():
+        shutil.rmtree(path)
+    path.mkdir(parents=True, exist_ok=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as config:
+        scrapper_config = json.load(config)
+
+    seed_urls = scrapper_config["seed_urls"]
+    max_articles = scrapper_config["total_articles_to_find_and_parse"]
+
+    if not isinstance(seed_urls, list):
+        raise IncorrectURLError
+    if not seed_urls:
+        raise IncorrectURLError
+    for seed_url in seed_urls:
+        if URL_PATTERN not in seed_url:
+            raise IncorrectURLError
+
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 300:
+        raise NumberOfArticlesOutOfRangeError
+
+    return seed_urls, max_articles
+
+
+class HTMLParser:
+    """
+    Parser implementation
+    """
+
+    def __init__(self, article_url, article_id):
+        """
+        Init
+        """
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(url=article_url, article_id=article_id)
+
+    def parse(self):
+        """
+        Parses each article
+        """
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+
+        table_bs = article_bs.find('div', class_="fulltext")
+        tag = table_bs.find('a')['href']
+        link = re.sub(r'(?i)view(?=\W)', 'download', tag)
+        pdf_file = PDFRawFile(link, self.article_id)
+
+        pdf_file.download()
+        self.article.text = pdf_file.get_text()
```

Comment:
read the docs carefully, we do not want references from a paper

Reply:
done